lxd/db/schema/query.go

```diff
@@ -0,0 +1,38 @@
+package schema
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/lxc/lxd/lxd/db/query"
+)
+
+// Return whether the schema table is present in the database.
+func querySchemaTableExists(tx *sql.Tx) (bool, error) {
+	rows, err := tx.Query(stmtSchemaTableExists)
+	if err != nil {
+		return false, err
+	}
+	defer rows.Close()
+	if !rows.Next() {
+		return false, fmt.Errorf("schema table query returned no rows")
+	}
+	var count int
+	if err := rows.Scan(&count); err != nil {
+		return false, err
+	}
+	return count == 1, nil
+}
+
+const stmtSchemaTableExists = `
+SELECT COUNT(name) FROM sqlite_master WHERE type = 'table' AND name = 'schema'
```

Comment:
The reason is purely aesthetic, i.e. to avoid having backticks break Go's regular indentation too badly.

Example:

```go
func foo(tx *sql.Tx) () {
        tx.Query(stmtFoo)
}

const stmtFoo = `
SELECT * FROM foo
    WHERE many='clauses are there'
        AND they='require backticks for readability'
`
```

vs:

```go
func foo(tx *sql.Tx) () {
       stmt = `
SELECT * FROM foo
    WHERE many='clauses are there'
        AND they='require backticks for readability'
`
        tx.Query(stmt)
}
```

I've changed this to inline all statements into their respective functions. Not sure if readability improved or worsened, I guess it depends on the reader. There's surely less indirection this way.